test-flows/README.md

```diff
@@ -1,41 +1,92 @@
 # Test Flows README
 
-To test this node, we use influxdb running in a docker container.  Currently testing with influxdb 1.6.3
+To test this node, we use InfluxDB running in a docker container. Currently tested with InfluxDB 1.8 and 2.0.
 
-Set up influxdb using docker.  See documentation at https://hub.docker.com/_/influxdb/
+Set up InfluxDB using docker. See documentation at https://hub.docker.com/_/influxdb/
 
-## Generating influxdb configuration
+## Generating InfluxDB 1.8 configuration
 
 We have a configuration file already set up for use with a self signed cert.  To generate a new, fresh config file locally:
 
     docker run --rm influxdb:1.8 influxd config > influxdb.conf
 
 ## Set up self signed certificate
 
-From the [influxdb admim documentation](https://docs.influxdata.com/influxdb/v1.6/administration/https_setup/) you can set up
+From the [InfluxDB admim documentation](https://docs.influxdata.com/influxdb/v1.8/administration/https_setup/) you can set up
 self signed SSL cert as follows:
 
     sudo openssl req -x509 -nodes -newkey rsa:2048 -keyout ./keys/influxdb-selfsigned.key -out ./keys/influxdb-selfsigned.crt
 
 Answer the questions as you like.
 
-## Running influxdb for tests
+## Running InfluxDB for tests
 
-To run influxdb using the config file in the current directory:
+### Docker Method
+
+To run InfluxDB 1.8 using the config file in the current directory:
 
     docker run --name=influxdb -p 8086:8086 \
       -v $PWD/influxdb.conf:/etc/influxdb/influxdb.conf:ro \
       -v $PWD/ssl:/etc/ssl \
       influxdb:1.8 -config /etc/influxdb/influxdb.conf
 
-To run the influxdb CLI against this container using a self-signed cert:
+To run the InfluxDB CLI against this container using a self-signed cert:
 
     docker run --rm --link=influxdb -it influxdb influx -ssl -unsafeSsl -host influxdb
 
-You can then execute CLI commands to create databases, make queries, etc..
+You can then execute CLI commands to create databases, make queries, etc.
 
-First, create a `test database to use by the test flows.  
+First, create a `test` database to use by the test flows.  
 
     create database test
 
-Then import the test flows into Node-RED and ensure they work.
\ No newline at end of file
+Then, import the test flows into Node-RED and ensure they work.
+
+### Docker Compose Method
+
+To run InfluxDB 1.8, 2.0 and Chronograf using the config file in the current directory:
+
+    docker-compose up
+
+Then, import the test flows into Node-RED and ensure they work.
```

Comment:
The tests using the 1.8 configuration work, but the 1.8 flux and 2.0 configurations report 'Error: self signed certificate'.  I don't see how to configure the TLS settings for 1.8 and 2.0 versions to get rid of this error.  Can we add TLS configuration for those versions?

Reply:
perhaps all we need to support self signed certificates is to add a flag.  Adding `rejectUnauthorized: false` to the client options seemed to work when using a self signed cert.

Reply:
I did not use the TLS settings for 1.8-flux and 2.0 since I did not see that config option in the new API. Instead of that, in the configuration URL we need to write the complete url with 'https' protocol when connecting to the InfluxDB server. The test flows should work out of the box when using the docker-compose framework. 

Did you follow the following and did not work?
1 - Launch Chronograf for InfluxDB 1.8 and 1.8 Flux: https://localhost:8888/

    Select Unsafe SSL option in the Connection Configuration section with Chronograf. Create a test database to be used by the test flows.
    Test flows parameters:
        Database: test
        Measurement: test
        Precision: ms
        Credentials:
            Username: username
            Password: password

2 - Launch Chronograf for InfluxDB 2.0: https://localhost:9999

    Test flows parameters:
        Bucket: test
        Measurement: test
        Precision: ms
        Organisation: my-org
        Credentials:
            Username: my-user
            Password: my-password
            Token: Create one under Load Data/Tokens

If this is the case I can test it in a clean environment.

Reply:
This means that the config nodes in the test-flows need to be modified before running the tests. You need to add the password for 1.8-flux and the token for 2.0

Reply:
Ups. I almost forgot. As mentioned in my first post:

- TLS standard Node-RED nodes are not used for the new nodes. Instead of that, the full URL has to be added to the configuration nodes. This means that to use SSL, Node-RED certs have to be properly configured. Otherwise, Node-RED has to be started up with the next command: NODE_TLS_REJECT_UNAUTHORIZED=0 node-red

Reply:
Yes, I tried with a clean environment, but I didn't run Node-RED with that environment variable.  That would work, but I think it would be better if the config node supported the 'rejectUnauthorized' flag so that the user can configure the client to use a self-signed certificate without setting the environment variable for the process.

I tried adding 'rejectUnauthorized:false' to the client options and  this seems to work.  This checkbox need to be documented as well.

Reply:
Thank you for the comments! Yes, a new checkbox should be created to allow this in the new 1.8 flux and 2.0 modes. Should the default rejectUnauthorized value be false?

Are you merging the current PR into your master and making the reviewed changes or do you prefer that I first implement and commit some of them into the current pull request?

Reply:
I think the default should be true since it's more secure by default that way.  If you don't mind making the changes in the current PR, I'd appreciate it.  Thanks!

Reply:
Sorry to recall this issue that late, I am a new user of InfluxDB. Congratulations for this project and for the adaptations to InfluxDB 2.0 (the one I am starting to use).
I am not too familiar with InfluxDB, but I am with NodeRED and https. However, I do not quite understand the way of using TLS with InfluxDB 2.0 configuration node.
I do not see how and where to specify the cert file and key files to use. The ones that are specified in nodered settings file is specific fot nodered web server to use secure connections with browsers
In my installation, nodered is not secured (its url is accessed with http) as it is in a LAN. But my infloxdb instance is in a remote server, secured, using TLS connections.
So, I do not see the way of configuring influxdb node to connect to that https remote influxdb server.
I would be very grateful if you can give some advice about this. I do not manage JavaScritp and node red nodes development as much as to be able to adapt the code, in case it is possible, and you said that TLS configuration is not implemented in the 2.0 API.
Thanks a lot!

Reply:
Finally, I have been able to connect using certificate and key using InfluxDB 2.0 configuration node.
I have done a hardcoded test, addíng two fields to clientOptions structure, in function InfluxConfigNode (influxdb.js), when creating a client connection (this.client = new InfluxDB(clientOptions)) in case influx version is 1.8 or 2.0:
fields 'key' and 'cert', with a strings to the paths of the corresponding files.
Also, it has been necessary to add at the beginning of the code:
const tls = require('node:tls');
const fs = require('node:fs');

I cannot tell why I have done these changes, but reading randomly tls in nodejs: https://nodejs.org/api/tls.html#tlsconnectoptions-callback

It seems that the structure accepts those two fields.

Now the next step would be to integrate it with a TLS config node? If you consider that is the correct way, I can try developing the code, although my knowledge on JavaScript and nodered nodes is limited, bu I can try